outline/electron/main.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	oss "github.com/Jigsaw-Code/outline-go-tun2socks/outline/shadowsocks"
+	"github.com/Jigsaw-Code/outline-go-tun2socks/shadowsocks"
+	"github.com/eycorsican/go-tun2socks/common/log"
+	"github.com/eycorsican/go-tun2socks/core"
+	"github.com/eycorsican/go-tun2socks/proxy/dnsfallback"
+	"github.com/eycorsican/go-tun2socks/tun"
+)
+
+const (
+	mtu        = 1500
+	udpTimeout = 30 * time.Second
+)
+
+var args struct {
+	tunAddr       *string
+	tunGw         *string
+	tunMask       *string
+	tunName       *string
+	tunDNS        *string
+	proxyHost     *string
+	proxyPort     *int
+	proxyPassword *string
+	proxyCipher   *string
+	logLevel      *string
+	dnsFallback   *bool
+	version       *bool
+}
+var version = ""
```

Comment:
I guess this should be nonempty? How are we going to keep this up to date?

We could use an approach like this: https://blog.alexellis.io/inject-build-time-vars-golang/

Reply:
We're using the `-X main.version=[version]` argument of `go`, allowing us to set the value at compile time.

Reply:
OK. In that case, I would make this `var version String` and add a comment indicating how the value is populated.

Reply:
Done.